src/diffusers/loaders.py

```diff
@@ -2276,6 +2436,93 @@ def enable_lora_for_text_encoder(self, text_encoder: Optional[PreTrainedModel] =
             raise ValueError("Text Encoder not found.")
         set_adapter_layers(self.text_encoder, enabled=True)
 
+    def set_adapters(
+        self,
+        adapter_names: Union[List[str], str],
+        unet_weights: List[float] = None,
+        te_weights: List[float] = None,
+        te2_weights: List[float] = None,
+    ):
+        # Handle the UNET
+        self.unet.set_adapters(adapter_names, unet_weights)
+
+        # Handle the Text Encoder
+        if hasattr(self, "text_encoder"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder, te_weights)
+        if hasattr(self, "text_encoder_2"):
+            self.set_adapters_for_text_encoder(adapter_names, self.text_encoder_2, te2_weights)
+
+    def disable_lora(self):
```

Comment:
Might make sense to have a reference to this method when `unload_lora_weights()` is called on a pipeline to make the users aware of the upcoming deprecation.

Reply:
Sorry I don't fully follow. Could you elaborate?

Reply:
I proposed something in [64e2d87](https://github.com/huggingface/diffusers/pull/5151/commits/64e2d87fa266dd0e69f580a5398fceaf1d6ce1c7) - let me know what do you think

Reply:
Resolving as https://github.com/huggingface/diffusers/pull/5151#issuecomment-1754657982